engines/freescape/freescape.h

```diff
@@ -0,0 +1,354 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef FREESCAPE_H
+#define FREESCAPE_H
+
+#include "common/bitarray.h"
+#include "common/random.h"
```

Comment:
This is not needed here. It is best to forward-declare as then it reduces header dependency. e.g.

```C++
namespace Common {
class RandomSource;
}
```

Etc. Most of these includes could be avoided. Then you add the `#include` directive to the actual .cpp file where it is used.

Reply:
I simplified or removed most of the includes, but I think I can use more foward declarations. In any case, you can take a look already.